src/Compilers/Core/Portable/SourceGeneration/GeneratorDriver.cs

```diff
@@ -183,6 +122,140 @@ public GeneratorDriver AddAdditionalTexts(ImmutableArray<AdditionalText> additio
             return FromState(newState);
         }
 
+        public GeneratorDriverRunResult GetRunResult()
+        {
+            var results = ArrayBuilder<GeneratorRunResult>.GetInstance(_state.Generators.Length);
```

Comment:
>ArrayBuilder<GeneratorRunResult>.GetInstance(_state.Generators.Length); [](start = 26, length = 71)

Consider:
```C#
var results = _state.Generators.ZipAsArray(
    _state.GeneratorStates,
    (generator, generatorState) => generatorState.SourceTexts.ZipAsArray(...));
```

Reply:
Not sure if its necessarily clearer what it's doing, but `ZipAsArray` has a bunch of small case optimizations (which will be common in this case), so I went with it. 

---
In reply to: [466747861](https://github.com/dotnet/roslyn/pull/46296#discussion_r466747861) [](ancestors = 466747861)